src/core/net/netif.cpp

```diff
@@ -122,8 +123,12 @@ bool Netif::IsMulticastSubscribed(const Address &aAddress) const
 {
     bool rval = false;
 
-    if (aAddress.IsLinkLocalAllNodesMulticast() || aAddress.IsRealmLocalAllNodesMulticast() ||
-        aAddress.IsRealmLocalAllMplForwarders())
+    if (mMulticastPromiscuousMode)
```

Comment:
I think we only want to apply the promiscuous mode to messages passed up to the host via `otReceiveIp6DatagramCallback`.  Applying the logic here will disable any multicast group filtering applied to Thread-internal multicast messages.